keel/api/views.py

```diff
@@ -1,3 +1,38 @@
-from django.shortcuts import render
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
+from .permissions import CustomLeadPermission
+from keel.leads.models import CustomerLead
+from keel.leads.serializers import CustomerLeadSerializer
 
-# Create your views here.
+class CustomerLeadView(APIView):
+
+    permission_classes = (CustomLeadPermission, )
+
+    def post(self, request, format="json"):
+        serializer = CustomerLeadSerializer(data=request.data)
+
+        if not serializer.is_valid():
+
+            data = {
+                "status" : 0,
+                "message" : serializer.errors
+            }
+            return Response(data, status=status.HTTP_400_BAD_REQUEST)
+        
+        custom_lead = serializer.save()
+
+        if not custom_lead:
```

Comment:
if the serialiser is valid...then there wont be any case where you wont find the custom lead object
if you want to get check on db operation then use a try except statement please

Reply:
done

Reply:
updated please check